### PR TITLE
Add thread name to DiagnosticEventTaskScheduler

### PR DESCRIPTION
**Pull request by ryzngard**
DiagnosticEventTaskScheduler currently does not name the threads, making worker threads from Roslyn hard to follow. This adds a thread name to the start callback for the long running thread. 

#29922

### CODE REVIEW

File: src/Features/Core/Portable/Diagnostics/DiagnosticEventTaskScheduler.cs [hunk 1]
```diff
@@ -27,6 +27,7 @@ public DiagnosticEventTaskScheduler(int blockingUpperBound)
 
         private void Start()
         {
+            Thread.CurrentThread.Name = "Roslyn Diagnostics";
```
Comment (heejaechang) [inline src/Features/Core/Portable/Diagnostics/DiagnosticEventTaskScheduler.cs]: can we just make it same pattern as the other one?

Comment (ryzngard) [reply src/Features/Core/Portable/Diagnostics/DiagnosticEventTaskScheduler.cs]: The other instances I saw start a thread directly instead of using the Task factory. We could transition to the same method here and name the created thread, but I wasn't sure that was the right approach.

Comment (heejaechang) [reply src/Features/Core/Portable/Diagnostics/DiagnosticEventTaskScheduler.cs]: let's go the other route since there we use both Thread and Task. I rather just use 1 (Thread or Task) rather mix them.
